python-checks/src/main/java/org/sonar/python/checks/UnusedLocalVariableCheck.java

```diff
@@ -120,17 +123,36 @@ private static boolean isVariableAccessedInStringTemplate(Symbol symbol, StringL
 
   public PreciseIssue createIssue(SubscriptionContext ctx, Symbol symbol, Usage usage) {
     if (isSequenceUnpacking(usage)) {
-      var quickFix = PythonQuickFix.newQuickFix(SEQUENCE_UNPACKING_QUICK_FIX_MESSAGE, TextEditUtils.replace(usage.tree(), "_"));
+      var quickFix = PythonQuickFix.newQuickFix(RENAME_QUICK_FIX_MESSAGE, TextEditUtils.replace(usage.tree(), "_"));
       var issue = ctx.addIssue(usage.tree(), String.format(SEQUENCE_UNPACKING_MESSAGE, symbol.name()));
       issue.addQuickFix(quickFix);
       return issue;
+    } else if (isLoopIndex(usage, symbol)) {
+      PreciseIssue issue = ctx.addIssue(usage.tree(), String.format(LOOP_INDEX_MESSAGE, symbol.name()));
+      PythonQuickFix quickFix = PythonQuickFix.newQuickFix(RENAME_QUICK_FIX_MESSAGE, TextEditUtils.replace(usage.tree(), "_"));
+      Symbol foundUnderscoreSymbol = null;
+      Tree searchTree = usage.kind().equals(Usage.Kind.LOOP_DECLARATION) ? ctx.syntaxNode() : null;
+      while (foundUnderscoreSymbol == null && searchTree != null) {
+        foundUnderscoreSymbol = ((FunctionDef) searchTree).localVariables().stream().filter(symbol1 -> "_".equals(symbol1.name())).findAny().orElse(null);
+        searchTree = TreeUtils.firstAncestorOfKind(searchTree, Kind.FUNCDEF);
```

Comment:
What if `_` is defined at the file input level or at the class level?

```
_ = 42
def foo():
  for i in range(5):  # quick fixing i into _ would change the behavior of the program
    print("hello")
  print(_)
foo()
```

I believe we wouldn't have this problem for a class, but we might want to have a test showing that this is the expected behavior.